h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java

```diff
@@ -0,0 +1,45 @@
+package water.api.schemas3;
+
+import water.H2ONode;
+import water.Iced;
+import water.fvec.Frame;
+import water.api.API;
+import water.fvec.Vec;
+
+public class FrameChunksV3 extends SchemaV3<Frame, FrameChunksV3> {
+
+    @API(help="ID of a given frame", json=false)
+    public KeyV3.FrameKeyV3 frame_id;
+
+    @API(help="Description of particular chunks", direction=API.Direction.OUTPUT)
+    public FrameChunkV3[] chunks;
+    
+    public static class FrameChunkV3 extends SchemaV3<Iced, FrameChunkV3> {
+
+        @API(help="An identifier unique in scope of a given frame", direction=API.Direction.OUTPUT)
+        public int chunk_id;
+
+        @API(help="Number of rows represented byt the chunk", direction=API.Direction.OUTPUT)
+        public int row_count;
+        
+        @API(help="Description of H2O node where the chunk is located", direction=API.Direction.OUTPUT)
+        public CloudV3.NodeV3 location;
```

Comment:
NodeV3 has quite a bit of information in it, this info will be replicated many times as many chunks will reside on the same node. I would use node index instead. WDYT?

Reply:
I like your idea, but I can't find anything like this on `NodeV3` is the node index equal to the position in `CloudeV3.nodes`?

Reply:
you can call `key.home().index()`

  // index of this node in the current cloud... can change at the next cloud.
  public int index() { return H2O.CLOUD.nidx(this); }

Reply:
I mentioned the position in `CloudV3.nodes` in order to be able to correctly map the node index to the correct `NodeV3` instance within the array. [It seems that Sparkling Water already relies on the equality of the node order within `CloudV3.nodes` and the node index.](https://github.com/h2oai/sparkling-water/blob/features/dynamic-allocation/core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala#L102)

Reply:
Yes, the order is given once the cloud is formed. You can rely on that.

Reply:
Cool, thanks!